src/main/java/com/google/api/codegen/viewmodel/DynamicLangXApiView.java

```diff
@@ -101,13 +101,25 @@ public boolean hasMissingDefaultOptions() {
   @Nullable
   public abstract String packageVersion();
 
+  @Nullable
+  public abstract List<String> validDescriptorsNames();
+
+  @Nullable
+  public abstract String constructorName();
+
+  public abstract boolean isGcloud();
+
   @Override
   public String resourceRoot() {
     return SnippetSetRunner.SNIPPET_RESOURCE_ROOT;
   }
 
   public static Builder newBuilder() {
-    return new AutoValue_DynamicLangXApiView.Builder();
+    return new AutoValue_DynamicLangXApiView.Builder().isGcloud(false);
+  }
+
+  public boolean hasValidDescriptorsNames() {
```

Comment:
This is unnecessary. Use `@if xapiClass.validDescriptorsNames` in the snippets instead.